### DISCUSSION TITLE
Support for 4 channels in SSIM

### DISCUSSION THREAD

**Discussion by kanishk-aidash** [francois-rozet/piqa#22] [Q&A] [answered]
Do we have support for 4 channels in SSIM ?

If I try to simple pass 4 channel images, it throws an error!

**Comment by francois-rozet** [comment]
Hi,

They should support it. Could you provide a minimal working example demonstrating the error?

**Comment by kanishk-aidash** [comment]
AssertionError: Number of channels expected to be 3, got 4

With both PSNR and SSIM



```
class SSIMLoss(SSIM):
    def forward(self, x, y):
        return 1. - super().forward(x, y)

ssim = SSIMLoss(value_range=255.0).cuda()

---

ssim(img1, img2)
```

img1 and img2 are 4 channel rasters

**Comment by francois-rozet** [comment, marked as answer]
As mentioned in the documentation, SSIM has a parameter `n_channels` which is 3 by default.

See https://francois-rozet.github.io/piqa/piqa.ssim.html#piqa.ssim.SSIM

**Comment by kanishk-aidash** [comment]
The SImilar issue is for PSNR!

```
class PSNRLoss(PSNR):
    def forward(self, x, y):
        return 100. - super().forward(x, y)

psnr = PSNRLoss(value_range=255.0).cuda()

---

psnr(img1, img2)
```

Gives a similar error, and I don't see the number of channels in its documentation! 

https://francois-rozet.github.io/piqa/piqa.psnr.html#piqa.psnr.PSNR

**Reply by francois-rozet** [reply]
Are you sure? There is no constraint for the number of channels in PSNR. I tested with various numbers, and no errors were raised.
